api/http/api_inventory_test.go

```diff
@@ -1526,6 +1526,84 @@ func TestUserAdmApiCreateTenant(t *testing.T) {
 	}
 }
 
+func TestApiInventoryInternalDevicesStatus(t *testing.T) {
+	t.Parallel()
+
+	rest.ErrorFieldName = "error"
+
+	testCases := map[string]struct {
+		inReq *http.Request
+
+		inventoryErr error
+
+		resp             utils.JSONResponseParams
+		deviceAttributes model.DeviceAttributes
+	}{
+		"ok": {
+			inReq: test.MakeSimpleRequest("POST",
+				"http://1.2.3.4/api/internal/v2/inventory/devices/:tenant_id/:status",
+				[]string{"1", "2", "3"},
+			),
+			resp: utils.JSONResponseParams{
+				OutputStatus: http.StatusOK,
+			},
+		},
+
+		"error, payload empty": {
+			inReq: test.MakeSimpleRequest("POST",
+				"http://1.2.3.4/api/internal/v2/inventory/devices/:tenant_id/:status",
+				nil,
+			),
+			resp: utils.JSONResponseParams{
+				OutputStatus:     http.StatusBadRequest,
+				OutputBodyObject: RestError("cant parse device ids: JSON payload is empty"),
+			},
+		},
+
+		"error, payload not expected": {
+			inReq: test.MakeSimpleRequest("POST",
+				"http://1.2.3.4/api/internal/v2/inventory/devices/:tenant_id/:status",
+				"unexpected here",
+			),
+			resp: utils.JSONResponseParams{
+				OutputStatus:     http.StatusBadRequest,
+				OutputBodyObject: RestError("cant parse device ids: json: cannot unmarshal string into Go value of type []string"),
+			},
+		},
+
+		"error, db Upsert failed": {
+			inReq: test.MakeSimpleRequest("POST",
+				"http://1.2.3.4/api/internal/v2/inventory/devices/:tenant_id/:status",
+				[]string{"12", "12", "12"},
+			),
+			inventoryErr: errors.New("cant upsert"),
+			resp: utils.JSONResponseParams{
+				OutputStatus:     http.StatusInternalServerError,
+				OutputBodyObject: RestError("internal error"),
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Logf("test case: %s", name)
+		inv := minventory.InventoryApp{}
+
+		ctx := contextMatcher()
+
+		inv.On("UpsertAttributes",
+			ctx,
+			mock.AnythingOfType("model.DeviceID"),
+			mock.AnythingOfType("model.DeviceAttributes"),
```

Comment:
I would rather implement func-based mocks that check the actual value of the deviceID and deviceAttributes you receive from the UpsertAttributes call.

Reply:
my [pleasure](https://github.com/mendersoftware/inventory/pull/162/commits/a71f3299d8788c8abb31838f631a53b0ac321f7b#diff-e00c71bbd3855c0ab6891e3a9d7a0dbbR1634)